api/turing/api/ensembling_job_api.go

```diff
@@ -0,0 +1,85 @@
+package api
+
+import (
+	"net/http"
+
+	mlp "github.com/gojek/mlp/client"
+	"github.com/gojek/turing/api/turing/models"
+	"github.com/gojek/turing/api/turing/service"
+)
+
+// EnsemblingJobController is the HTTP controller that handles the orchestration of ensembling jobs.
+type EnsemblingJobController struct {
+	BaseController
+}
+
+// Create is a HTTP Post Method that creates an Ensembling job.
+// This method will only return the acceptance/rejection of the job rather than the actual result of the job.
+func (c EnsemblingJobController) Create(
+	r *http.Request,
+	vars RequestVars,
+	body interface{},
+) *Response {
+	var errResp *Response
+	var project *mlp.Project
+	if project, errResp = c.getProjectFromRequestVars(vars); errResp != nil {
+		return errResp
+	}
+
+	// Check is done in api.handlers
+	request, _ := body.(*models.EnsemblingJob)
+	request.ProjectID = models.ID(project.Id)
+
+	// Check if ensembler exists
+	ensembler, err := c.EnsemblersService.FindByID(
+		request.EnsemblerID,
+		service.EnsemblersFindByIDOptions{
+			ProjectID: &request.ProjectID,
+		},
+	)
+	if err != nil {
+		return NotFound("ensembler not found", err.Error())
+	}
+
+	// Populate name if the user does not define a name for the job
+	if request.Name == "" {
+		request.Name = c.EnsemblingJobService.GenerateDefaultJobName(
+			ensembler.GetName(),
+		)
+	}
+
+	// Populate default environment
+	request.EnvironmentName = c.EnsemblingJobService.GetDefaultEnvironment()
+
+	// Populate ensembler directory
+	ensemblerDirectory, err := c.EnsemblingJobService.GetEnsemblerDirectory(ensembler)
+	if err != nil {
+		return BadRequest("ensembler not supported", err.Error())
+	}
+	request.EnsemblerConfig.EnsemblerConfig.Spec.Ensembler.Uri = ensemblerDirectory
+
+	// Populate ensembler artifact URI, error can be ignored since the type check is done prior
+	artifactURI, _ := c.EnsemblingJobService.GetArtifactURI(ensembler)
+	request.InfraConfig.ArtifactURI = artifactURI
+	request.InfraConfig.EnsemblerName = ensembler.GetName()
```

Comment:
All this logic should be done inside the `EnsemblingJobService`, not at the controller level. You just need to pass an instance of `ensembler` into the relevant method of that service and populate all the necessary fields from it inside of that method

Reply:
Ref: https://github.com/gojek/merlin/blob/8f52ce14c73de3a3efcafb9a9bca40138ee38c89/api/api/prediction_job_api.go#L58

Reply:
done